tests/gadget-zero/usb-gadget0.c

```diff
@@ -297,6 +300,37 @@ static int gadget0_control_request(usbd_device *usbd_dev,
 			*len = req->wValue;
 		}
 		return USBD_REQ_HANDLED;
+	case GZ_REQ_WRITE_LOOPBACK_BUFFER:
+		if (req->wValue > sizeof(usbd_control_buffer)) {
+			ER_DPRINTF("Can't store more than out control buffer! %d > %d\n",
+				req->wValue, sizeof(usbd_control_buffer));
+			return USBD_REQ_NOTSUPP;
+		}
+		/* Don't store more than asked for! */
+		if (req->wValue > req->wLength) {
```

Comment:
I wouldn't have thought this was a necessary check, as long as you didn't overflow, but I guess it doesn't hurt.